Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -124,12 +124,19 @@ JSValue eval(CallFrame* callFrame, JSValue thisValue, JSScope* callerScopeChain,
 
     JSValue program = callFrame->argument(0);
     JSString* programString = nullptr;
+    bool isTrusted = false;
     if (LIKELY(program.isString()))
         programString = asString(program);
     else if (Options::useTrustedTypes()) {
-        auto code = globalObject->globalObjectMethodTable()->codeForEval(globalObject, program);
-        if (!code.isNull())
-            programString = jsString(vm, code);
+        auto* structure = globalObject->trustedScriptStructure();
+        if (!structure) {
+            structure = globalObject->globalObjectMethodTable()->trustedScriptStructure(globalObject);
```

Comment:
This should have an UNLIKELY()

@Constellation  Is it faster to do this in the global object constructor instead?

Reply:
I've added the UNLIKELY

Reply:
Yeah, do that in global object constructor instead.